sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShuffleBufferCatalog.scala

```diff
@@ -16,52 +16,77 @@
 
 package com.nvidia.spark.rapids
 
-import java.io.File
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.atomic.AtomicInteger
 import java.util.function.{Consumer, IntUnaryOperator}
 
 import scala.collection.mutable.ArrayBuffer
 
-import ai.rapids.cudf.{ContiguousTable, Cuda, DeviceMemoryBuffer}
+import ai.rapids.cudf.{ContiguousTable, Cuda, DeviceMemoryBuffer, Table}
 import com.nvidia.spark.rapids.Arm.withResource
 import com.nvidia.spark.rapids.format.TableMeta
+import com.nvidia.spark.rapids.spill.{SpillableDeviceBufferHandle, SpillableHandle}
 
-import org.apache.spark.SparkEnv
+import org.apache.spark.{SparkEnv, TaskContext}
 import org.apache.spark.internal.Logging
-import org.apache.spark.sql.rapids.RapidsDiskBlockManager
 import org.apache.spark.sql.rapids.execution.TrampolineUtil
+import org.apache.spark.sql.types.DataType
+import org.apache.spark.sql.vectorized.ColumnarBatch
 import org.apache.spark.storage.ShuffleBlockId
 
 /** Identifier for a shuffle buffer that holds the data for a table */
 case class ShuffleBufferId(
     blockId: ShuffleBlockId,
-    override val tableId: Int) extends RapidsBufferId {
+    tableId: Int) {
   val shuffleId: Int = blockId.shuffleId
   val mapId: Long = blockId.mapId
-
-  override val canShareDiskPaths: Boolean = true
-
-  override def getDiskPath(diskBlockManager: RapidsDiskBlockManager): File = {
-    diskBlockManager.getFile(blockId)
-  }
 }
 
 /** Catalog for lookup of shuffle buffers by block ID */
-class ShuffleBufferCatalog(
-    catalog: RapidsBufferCatalog,
-    diskBlockManager: RapidsDiskBlockManager) extends Logging {
+class ShuffleBufferCatalog extends Logging {
+  /**
+   * Information stored for each active shuffle.
+   * NOTE: ArrayBuffer in blockMap must be explicitly locked when using it!
+   *
+   * @param blockMap mapping of block ID to array of buffers for the block
```

Comment:
blockMap appears in the file, but the diff doesn't contain lines that touch it (they are there if you open the whole file). 

It's a param because it's a case class. I think we tend to use `@param` in this case.

I will update the comment, it should really say "mapping of blockID to array of shuffle buffer IDs for the block". This concept I didn't change in the PR, but I did remove code from this file.

Reply:
never mind. I had the wrong code locally. You are right, removing it.  I kept looking through the code thinking "I'd like to remove all of this stuff.. why is it still here" => I had already removed it